Common/include/element_structure.hpp

```diff
@@ -415,48 +392,45 @@ class CElement {
 
   /*!
    * \brief Compute the value of the area of the element
+   * \param[in] mode - Type of coordinates to consider in the computation
    * \param[out] val_Area - Area of the element
    */
-  virtual su2double ComputeArea(void);
+  virtual su2double ComputeArea(const FrameType mode = REFERENCE);
 
   /*!
    * \brief Compute the value of the volume of the element
+   * \param[in] mode - Type of coordinates to consider in the computation
    * \param[out] val_Volume - Volume of the element
    */
-  virtual su2double ComputeVolume(void);
+  virtual su2double ComputeVolume(const FrameType mode = REFERENCE);
 
   /*!
-   * \brief Compute the value of the area of the element in current coordinates
+   * \brief Compute the value of the area of the element in current coordinates (wrapper to ComputeArea(CURRENT)).
    * \param[out] val_Area - Area of the element
    */
-  virtual su2double ComputeCurrentArea(void);
+  su2double ComputeCurrentArea(void);
```

Comment:
This is now just short-hand to calling ComputeArea with "CURRENT" as argument, this keeps the change from spreading to other areas of the code but one could always drop the ComputeCurrent* methods...

Reply:
Yep that's right

Reply:
For the mesh solver it would be very efficient to compute the volume the FE way (sum Ji*Wi) since the gradients need to be computed anyway.

Reply:
Yeah I also think so. I hope to be able to open the PR on the mesh solver soon(ish - next month and a half I will be mostly travelling), and maybe then we can give it a shot.